### ISSUE TITLE
Raspberry Pi Docker Edgerouter issues

### ISSUE THREAD

**Issue by andymarden** [Nico640/docker-unms#34]
Hi - been trying to get this configured for ages - everything is up and running in a docker container. I have INGINX using ports 80 and 443 so tried with:

Using Portainer, so set env variable:
WS_PORT 443
PUBLIC_WS_PORT 32770
PUBLIC_HTTPS_PORT 32770

Mapped Ports:
32770 > 443 (tcp)
2055 > 2055 (udp)

I see all the devices - when I try to add my EdgeRouter, it authenticates, updates the UNMS config in the router with, what I think is, the correct stuff for the IP address of my Pi:

wss://192.168.0.173:32770+etc...

But then sits there saying "waiting for..." (can't see the rest of the message)

I tried stopping NGINX and leaving ports as default, mapping 80, 443 and 2055 to the same, but still  the same behaviour.

Is there something about the EdgeRouter communicating back with the UNMS container? I know 443 gets through normally because NGINX uses it. Losing the will to live trying to get this working! So close and yet...

Cheers
Andy

**Comment by Nico640** [comment]
I don't have an EdgeRouter on hand right now, but I have an EdgePoint (similar firmware) running with no issues.
I tested your configuration (port 32770) with a LiteBeam and it connected just fine. Do I understand  you correctly that you only have this issue with the EdgeRouter? Do you have any other EdgeOS devices that work?

You could also try to copy the UNMS key (Devices->Add Device->Copy UNMS key) and enter it directly in the EdgeRouter web panel. Because you're using a private IP address, the key should have something like "+allowUntrustedCertificate" at the end.
Is your EdgeRouter somehow in a different environment than the other devices (other network) and is it able to reach the UNMS server?

**Comment by Danabw** [comment]
I'm using this wi/a ER 12, don't have any issues connecting to my ER and using UNMS w/it. 
Have you tried disabling/re-enabling UNMS in the System tab inthe ER settings, and then trying to reconnect? 

![2020-05-24 09_42_21-EdgeOS - Riggles](https://user-images.githubusercontent.com/24300160/82759554-107bcb80-9da3-11ea-9118-88cd02842e24.jpg)

**Comment by andymarden** [comment]
The Ubiquiti devices I have are the EdgeRouter PoE, the AP Lite and the Outdoor Mesh AP.

I tried at the start to enter the UNMS key in the router config - basically whether I do or don't, when I click ADD in UNMS, it asks for the router username and password, the sets the unms settings itself, overwriting anything that is there and turning on UNMS if it is off. The EdgeRouter System page says teh Config has changed and I have to reload the page.

At the end of the URL it says "+allowSelfSignedCertificate" rather than "+allowUntrustedCertificate".

The EdgeRouter is my gateway at 192.168.01 internally - all devices are on 192.168.0.x network and docker containers on the RPi are, of course, on the 172.x.x.x private network.

OK - so I thought I would try the "+allowUntrustedCertificate" myself - so I started the UNMS docker image and saw the Add Device button top right. Filled in the IP address, user password. Then it shows 192.168.0.1 as an extra device, alongside ubnt. Then 192.168.0.1 disappeared and after a few seconds so did ubnt but is has now been successfully added to UNMS!!!!

I notice that in the EdgeRouter, the suffix of the UNMS URL is now  "+allowUntrustedCertificate" - but I didn't do anything apart from try to add the device again manually!

WTF?!? AT this stage I am thinking voodoo, astrology or the paranormal! It looks like adding the devise again somehow triggered it to use "+allowUntrustedCertificate".

Absolutely bizarre - but thanks - it's working now. Something weird is going taht woudl be better if it got it right the first time.
